sshd-core/src/main/java/org/apache/sshd/common/session/helpers/AbstractSession.java

```diff
@@ -1757,14 +1764,18 @@ protected void receiveNewKeys() throws Exception {
                     this, inCipher, outCipher, recommendedByteRekeyBlocks, maxRekeyBlocks);
         }
 
+        resetCounters();
+        lastKeyTimeValue.set(Instant.now());
+        firstKexPacketFollows = null;
+    }
+
+    protected void resetCounters() {
         inBytesCount.set(0L);
         outBytesCount.set(0L);
         inPacketsCount.set(0L);
         outPacketsCount.set(0L);
         inBlocksCount.set(0L);
         outBlocksCount.set(0L);
-        lastKeyTimeValue.set(Instant.now());
-        firstKexPacketFollows = null;
```

Comment:
Why were these deleted ? They are used in the code...

Reply:
Not deleted, just moved a little bit. This method keeps hitting the 100 line limit from checkstyle, so I had to refactor it a little.

Reply:
If it becomes too cumbersome you can use `@SuppressWarnings("checkstyle:...the rule")` - I do not encourage it, but sometimes it is better than the alternative

Reply:
It's not too cumbersome, though this particular method is justifiably being flagged by checkstyle. It's exactly 100 lines, and every SSH extension that doesn't follow the normal key exchange process (e.g., this chacha extension, and to some extent even the aes/gcm one) complicates this code further. Ideally, some form of strategy APIs could be devised around these areas, but it's possible that _what_ strategies are useful won't be apparent until more extensions are supported or created.

As for one potential related simplification to do in this method might be to extract some sort of data class to handle the c2s and s2c cipher/mac/IV/etc. data. Of course, that can always be done later.

Reply:
I've added a checkstyle suppression for now. My ideas on how this class could be refactored are a bit larger and orthogonal to this PR, so I added a TODO comment. My refactoring idea would be essentially to encapsulate the incoming and outgoing fields into a pair of instances of a sort of session data class.